module/sockets/grant-effect.mjs

```diff
@@ -0,0 +1,68 @@
+/**
```

Comment:
Seems like we could stick this into a class to formalize the interface a bit. Perhaps an abstract `SocketEvent` class that defines placeholder `name` property and abstract `initiative` and `finalize` methods.